src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -78,16 +78,25 @@ internal static OtlpLogs.LogRecord ToOtlpLog(this LogRecord logRecord)
                 otlpLogRecord.Body = new OtlpCommon.AnyValue { StringValue = logRecord.FormattedMessage };
             }
 
+            if (logRecord.StateValues != null)
+            {
+                foreach (var stateValue in logRecord.StateValues)
+                {
+                    var otlpAttribute = stateValue.ToOtlpAttribute();
+                    otlpLogRecord.Attributes.Add(otlpAttribute);
```

Comment:
Two thoughts regarding the transform here:

1. We may want to consider what happens when a state value's key is null/empty?
2. Prefixing attributes for state values with `state.` would make it clear that these attributes were sources from the state.

We have a similar component here https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/897b734aa5ea9992538f04f6ea6871fe211fa903/src/OpenTelemetry.Contrib.Preview/Internal/DefaultLogStateConverter.cs#L31-L42 that does this in the context of adding state values to an activity event.

The console exporter is another good reference point. It logs out state values under it's own section with the values indented:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/405bbda121e1e64e06daf3ba673bcebc425c57fc/src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs#L54-L58